src/renderer/atlas/AtlasEngine.api.cpp

```diff
@@ -615,28 +624,28 @@ void AtlasEngine::_resolveFontMetrics(const wchar_t* requestedFaceName, const Fo
     u16 glyphIndex;
     THROW_IF_FAILED(fontFace->GetGlyphIndicesW(&codePoint, 1, &glyphIndex));
 
-    DWRITE_GLYPH_METRICS glyphMetrics;
+    DWRITE_GLYPH_METRICS glyphMetrics{};
     THROW_IF_FAILED(fontFace->GetDesignGlyphMetrics(&glyphIndex, 1, &glyphMetrics));
 
     // Point sizes are commonly treated at a 72 DPI scale
     // (including by OpenType), whereas DirectWrite uses 96 DPI.
     // Since we want the height in px we multiply by the display's DPI.
-    const auto fontSizeInDIP = fontSize / 72.0f * 96.0f;
-    const auto fontSizeInPx = fontSize / 72.0f * _api.dpi;
-
-    const auto designUnitsPerPx = fontSizeInPx / static_cast<float>(metrics.designUnitsPerEm);
-    const auto ascent = static_cast<float>(metrics.ascent) * designUnitsPerPx;
-    const auto descent = static_cast<float>(metrics.descent) * designUnitsPerPx;
-    const auto lineGap = static_cast<float>(metrics.lineGap) * designUnitsPerPx;
-    const auto underlinePosition = static_cast<float>(-metrics.underlinePosition) * designUnitsPerPx;
-    const auto underlineThickness = static_cast<float>(metrics.underlineThickness) * designUnitsPerPx;
-    const auto strikethroughPosition = static_cast<float>(-metrics.strikethroughPosition) * designUnitsPerPx;
-    const auto strikethroughThickness = static_cast<float>(metrics.strikethroughThickness) * designUnitsPerPx;
-    const auto advanceWidth = static_cast<float>(glyphMetrics.advanceWidth) * designUnitsPerPx;
+    const auto dpi = static_cast<f32>(_api.s->font->dpi);
+    const auto fontSizeInPx = fontSize / 72.0f * dpi;

```

Comment:
By using `D2D1_UNIT_MODE_PIXELS` all of AtlasEngine now consistently operates in pixels, so there's no need for a `fontSizeInDIP` anymore.

Reply:
It's possible I missed something, but I don't think there are any such bugs in this code. Points can be inferred from the pixel font size via `FontSettings::dpi` at any time though.